### DISCUSSION TITLE
Default app role for all users

### DISCUSSION THREAD

**Discussion by mr-falken** [Budibase/budibase#5617] [Help]
Hi,
  first of all kudos to you guys for developing such a great platform: we're planning to move all our powerapp/office 365 automation tools in a self hosted budibase environment, it's really amazing how much it's powerful AND simple to use.

Back to the topic.. :)

For what i've understood, at the moment we need to defined access for an app one user at a time, editing the role in the user detail page as an admin.

Since we're integrating bb with our openid connect, I'd like to find a way to specify from the app something like "give all our users basic access to this app".

Is that possibile via GUI/REST api ?

The silver bullet would be group management, but I think that's something that's on the road to be implemented.

**Comment by shogunpurple** [comment]
Hi @mr-falken - first of all thank you very much for the kind words! We really appreciate it. Glad to hear you are planning a big move from your powerapps automation to budibase.

Yep, that's correct - we do have some things on our roadmap that we are calling "User Management V2" and "workspaces" which will seek to solve a lot of these problems, such as:
- Group role assignment of users
- AD Active directory snc
- A whole new user management UI

We've had quite a few requests for this functionality specifically. You can view the roadmap here:
https://github.com/Budibase/budibase/projects/10

We have the user management work scheduled for Q3, so the next few months.

Are there any other features you would like to see implemented as part of the user management piece?

Right now, you can do this through the budibase REST API, and specifically the user endpoint. You can see the documentation for that here:

https://docs.budibase.com/reference/post_users

In the create user endpoint, you can specify the `roles` under that user, giving them access to a bunch of apps. The format is like this:
```js
{
  email: "someemail@email.com",
  roles: {
    "appId": "BASIC"
  }
}
```

When you `POST` that to our API, it will create those users and assign them roles to the app you specify. You could write a script or a small frontend that would allow you to do this - you could even do it inside budibase using our REST connector.

Let me know if you have any more questions, and I'm happy to walk you through whatever issues you need help with.

Thank you!
